src/assets/js/admin/promotions.js

```diff
@@ -0,0 +1,155 @@
+jQuery( document ).ready( function( $ ) {
+
+
+	$.JiltPromotions = {}
+
+
+	/**
+	 * Jilt Plugin Installation modal handler.
+	 *
+	 * @since 1.1.0-dev.1
+	 */
+	$.JiltPromotions.InstallPluginModal = class InstallPluginModal {
+
+
+		/**
+		 * Constructor.
+		 *
+		 * @since 1.1.0-dev.1
+		 *
+		 * @param {object} options
+		 */
+		constructor( options ) {
+
+			this.options = {
+				messageID : options.messageID || '',
+				target    : options.target    || '',
+				onClose   : options.onClose   || '',
+			};
+
+			if ( '' === this.options.messageID ) {
+				console.log( 'InstallPluginModal: missing messageID' )
+				return;
+			}
+
+			if ( '' === this.options.target ) {
+				this.options.target = 'sv-wc-jilt-promotions-' + this.options.messageID + '-modal';
+			}
+
+			this.initialize();
+			this.open();
+		}
+
+
+		/**
+		 * Initializes the modal.
+		 *
+		 * @since 1.1.0-dev.1
+		 */
+		initialize() {
+
+			// when the install button is clicked
+			$( '#sv-wc-jilt-install-button-install' ).on( 'click', function( event ) {
+				InstallPluginModal.onInstall( event )
```

Comment:
I had to switched them to use arrow functions to access `this` easily.

Also, since the modal is not opened at this point, I had to set the event handlers on the document.

```suggestion
			$( document ).on( 'click', '#sv-wc-jilt-install-button-install', ( event ) => {
				this.onInstall( event );
```